app/src/main/java/com/example/wrk/fragments/ProfileFragment.java

```diff
@@ -372,4 +412,92 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    private void launchCamera() {
+        // create Intent to take a picture and return control to the calling application
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create a File reference for future access
+        photoFile = getPhotoFileUri(photoFileName);
+
+        // wrap File object into a content provider, required for API >= 24
+        Uri fileProvider = FileProvider.getUriForFile(getContext(), "com.wrk.fileprovider", photoFile);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
+
+        // So as long as the result is not null, it's safe to use the intent.
+        if (intent.resolveActivity(getContext().getPackageManager()) != null) {
+            // Start the image capture intent to take photo
+            startActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
+        }
+    }
+
+    // Trigger gallery selection for a photo
+    private void onPickPhoto(View view) {
+        // Create intent for picking a photo from the gallery
+        Intent intent = new Intent(Intent.ACTION_PICK,
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
```

Comment:
I have an impression maybe you referenced this code from some tutorial and kept all the comments? in that case, might be helpful to reassess if all of them are needed. if that is not the case and you just added comments yourself for clarity, then never mind